libdnf-cli/utils/tty.cpp

```diff
@@ -0,0 +1,91 @@
+#include "tty.hpp"
+
+#include <sys/ioctl.h>
+#include <termios.h>
+#include <unistd.h>
+
+#include <cstdlib>
+
+
+namespace libdnf::cli::utils::tty {
+
+
+int get_width() {
+    // Use a custom "FORCE_COLUMNS" variable for testing purposes.
+    // "COLUMNS" is overwritten in a sub-shell and that makes testing more difficult
+    char * columns = std::getenv("FORCE_COLUMNS");
+    if (columns != nullptr) {
+        try {
+            return std::stoi(columns);
+        } catch (std::invalid_argument & ex) {
+        } catch (std::out_of_range & ex) {
+        }
+    }
+
+    struct winsize size;
+    if (ioctl(STDOUT_FILENO, TIOCGWINSZ, &size) == 0) {
+        return size.ws_col;
+    }
+
+    return 80;
+}
+
+
+bool is_interactive(void) {
+    return isatty(fileno(stdout)) == 1;
+}
+
+
+#define TTY_COMMAND(name, color)                 \
```

Comment:
What is meaning of `color`? Is `reset`, `underline`or `blank` a color?
It seems like ANSI escape code (https://en.wikipedia.org/wiki/ANSI_escape_code).
I think `code`, `ansi_code`, `escape_code` names are more relevant then `color`.

Reply:
used `escape_code`